src/Mod/CAM/Path/Machine/models/machine.py

```diff
@@ -0,0 +1,1474 @@
+            "feed_precision": self.output.feed_precision,
+            "spindle_decimals": self.output.spindle_decimals,
+            "command_space": self.output.command_space,
+            "comment_symbol": self.output.comment_symbol,
+            "line_increment": self.output.line_increment,
+            "line_number_start": self.output.line_number_start,
+            "end_of_line_chars": self.output.end_of_line_chars,
+        }
+
+        # G-code blocks (only non-empty ones)
+        blocks = {}
+        if self.blocks.pre_job:
+            blocks["pre_job"] = self.blocks.pre_job
+        if self.blocks.post_job:
+            blocks["post_job"] = self.blocks.post_job
+        if self.blocks.preamble:
+            blocks["preamble"] = self.blocks.preamble
+        if self.blocks.postamble:
+            blocks["postamble"] = self.blocks.postamble
+        if self.blocks.safetyblock:
+            blocks["safetyblock"] = self.blocks.safetyblock
+        if self.blocks.pre_operation:
+            blocks["pre_operation"] = self.blocks.pre_operation
+        if self.blocks.post_operation:
+            blocks["post_operation"] = self.blocks.post_operation
+        if self.blocks.pre_tool_change:
+            blocks["pre_tool_change"] = self.blocks.pre_tool_change
+        if self.blocks.post_tool_change:
+            blocks["post_tool_change"] = self.blocks.post_tool_change
+        if self.blocks.tool_return:
+            blocks["tool_return"] = self.blocks.tool_return
+        if self.blocks.pre_fixture_change:
+            blocks["pre_fixture_change"] = self.blocks.pre_fixture_change
+        if self.blocks.post_fixture_change:
+            blocks["post_fixture_change"] = self.blocks.post_fixture_change
+        if self.blocks.pre_rotary_move:
+            blocks["pre_rotary_move"] = self.blocks.pre_rotary_move
+        if self.blocks.post_rotary_move:
+            blocks["post_rotary_move"] = self.blocks.post_rotary_move
+
+        if blocks:
+            data["blocks"] = blocks
+
+        # Processing options
+        data["processing"] = {
+            "modal": self.processing.modal,
+            "translate_drill_cycles": self.processing.translate_drill_cycles,
+            "split_arcs": self.processing.split_arcs,
+            "show_editor": self.processing.show_editor,
+            "list_tools_in_preamble": self.processing.list_tools_in_preamble,
+            "show_machine_units": self.processing.show_machine_units,
+            "show_operation_labels": self.processing.show_operation_labels,
+            "tool_before_change": self.processing.tool_before_change,
+            "drill_cycles_to_translate": self.processing.drill_cycles_to_translate,
+            "suppress_commands": self.processing.suppress_commands,
+            "tool_change": self.processing.tool_change,
+            "adaptive": self.processing.adaptive,
+            "chipbreaking_amount": self.processing.chipbreaking_amount,
```

Comment:
Why is this defined at the Machine level?   Isn't this a property of an operation?   Isn't it likely to vary from one op to another?